### PR TITLE
JCLOUDS-1183 oneandone-monitoringpolicy-api

### PR DESCRIPTION
**Pull request by alibazlamit**
@nacx moving this branch before the monitoring center, please review.

Thanks,

### CODE REVIEW

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/MonitoringPolicy.java [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+   public abstract String name();
+
+   @Nullable
+   public abstract String description();
+
+   @Nullable
+   public abstract String state();
+
+   @Nullable
+   public abstract String creationDate();
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/domain/MonitoringPolicy.java]: Change type to `Date`?

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+@Path("/monitoring_policies")
+@Produces("application/json")
+@Consumes("application/json")
+@RequestFilters(AuthenticateRequest.class)
+public interface MonitoringPolicyApi {
+
+   @Named("monitoringpolicy:list")
+   @GET
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<MonitoringPolicy> getList();
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java]: Renamt to `list()`.

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+   @Named("monitoringpolicy:list")
+   @GET
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<MonitoringPolicy> getList(GenericQueryOptions options);
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java]: Same here, rename to `list`.

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+   @Path("/{policyId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @MapBinder(BindToJsonPayload.class)
+   MonitoringPolicy delete(@PathParam("policyId") String policyId);
+
+   @Named("monitoringpolicy:ports:list")
+   @GET
+   @Path("/{policyId}/ports")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<MonitoringPolicy.Port> getPorts(@PathParam("policyId") String policyId);
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java]: Rename to `listPorts`

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+   @Path("/{policyId}/ports/{portId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @MapBinder(BindToJsonPayload.class)
+   MonitoringPolicy deletePort(@PathParam("policyId") String policyId, @PathParam("portId") String portId);
+
+   @Named("monitoringpolicy:processes:list")
+   @GET
+   @Path("/{policyId}/processes")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<MonitoringPolicy.Process> getProcesses(@PathParam("policyId") String policyId);
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java]: Rename to `listProcesses`

File: oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+   @Path("/{policyId}/processes/{processId}")
+   @Fallback(Fallbacks.NullOnNotFoundOr404.class)
+   @MapBinder(BindToJsonPayload.class)
+   MonitoringPolicy deleteProcess(@PathParam("policyId") String policyId, @PathParam("processId") String processId);
+
+   @Named("monitoringpolicy:servers:list")
+   @GET
+   @Path("/{policyId}/servers")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   List<MonitoringPolicy.Server> getServers(@PathParam("policyId") String policyId);
```
Comment (nacx) [inline oneandone/src/main/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApi.java]: Rename to `listServers`

File: oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApiLiveTest.java [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+
+   @Test(dependsOnMethods = "testUpdate")
+   public void testAddPort() throws InterruptedException {
+      List<MonitoringPolicy.Port.AddPort> ports = new ArrayList<MonitoringPolicy.Port.AddPort>();
+      MonitoringPolicy.Port.AddPort port = MonitoringPolicy.Port.AddPort.create(80, Types.AlertIfType.RESPONDING, true, Types.ProtocolType.TCP);
+      ports.add(port);
+      MonitoringPolicy response = monitoringPolicyApi().addPort(currentPolicy.id(), MonitoringPolicy.Port.CreatePort.create(ports));
+      currentPort = response.ports().get(0);
+
+      assertNotNull(response);
```
Comment (nacx) [inline oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApiLiveTest.java]: Assert before reading the first port

Comment (alibazlamit) [inline oneandone/src/test/java/org/apache/jclouds/oneandone/rest/features/MonitoringPolicyApiLiveTest.java]: @nacx Updated, let me know when to squash.

**Comment by alibazlamit** [comment]
@nacx Fixed and squashed.

**Comment by nacx** [comment]
Merged to master as [87c54700](http://git-wip-us.apache.org/repos/asf/jclouds-labs/commit/87c54700).

**Pull request closed without merging**